app/lib/actions/katello/upstream_subscriptions/remove_entitlements.rb

```diff
@@ -3,17 +3,26 @@ module Katello
     module UpstreamSubscriptions
       class RemoveEntitlements < Actions::Base
         def plan(pool_ids = [])
+          organization = Object::Organization.current
+
+        begin
+          ::Katello::Resources::Candlepin::UpstreamConsumer.get(organization)
+        rescue RestClient::Gone
+          raise ::Katello::Errors::UpstreamConsumerGone
+        end
+
           ids = pool_ids.uniq.compact
           fail _("No pool IDs were provided.") if ids.blank?
           fail _("Current organization is not set.") unless ::Organization.current
 
           sequence do
             ids.each do |pid|
               pool = ::Katello::Pool.find(pid)
+              sub_name = pool.subscription.name
```

Comment:
this assignment can probably be moved to below the 'fail' since in that situation it's a meaningless assignment :)